Source/options.cpp

```diff
@@ -1478,20 +1520,31 @@ std::vector<OptionEntryBase *> ModOptions::GetEntries()
 	return optionEntries;
 }
 
+void ModOptions::AddModEntry(const std::string &modName)
+{
+	auto &entries = GetModEntries();
+	entries.emplace_front(modName);
+}
+
+void ModOptions::RemoveModEntry(const std::string &modName)
+{
+	if (!modEntries) {
+		return;
+	}
+
+	auto &entries = *modEntries;
+	entries.remove_if([&](const ModEntry &entry) {
+		return entry.name == modName;
+	});
+}
+
 std::forward_list<ModOptions::ModEntry> &ModOptions::GetModEntries()
 {
 	if (modEntries)
 		return *modEntries;
 
 	std::vector<std::string> modNames = ini->getKeys(key);
 
-	// Add mods available by default:
-	for (const std::string_view modName : { "clock" }) {
-		if (c_find(modNames, modName) != modNames.end()) continue;
-		ini->set(key, modName, false);
-		modNames.emplace_back(modName);
-	}

```

Comment:
I see you removed the clock mod but provided no mechanism to add it back. Rather, even if I added it to my ini file manually, seems like your PR would remove it. Were you envisioning a way in which these different mechanisms could coexist?

FYI, I had plans to introduce a mechanism for placing an MPQ in the mods folder as well. So there would be three mechanisms: mod folders, mod MPQs, and packaged MPQs in `devilutionx.mpq`.

Perhaps we could call `entries.remove_if()` using `FindAsset()` instead of checking against the directory listing.

Reply:
Yeah build in mods needs to be "hard coded" some how since we can't discover content of the devilutionx.mpq

Solutions:
- Add a mod list file to the MPQ
- Hard code them in the exe like they where
- Add them as separate mods/files

Reply:
I'll give implementing all 3 methods a quick shot